### PR TITLE
#263 google meet creation implemented

### PR DESCRIPTION
**Pull request by YuraRiabov**
Screenshot_785
Screenshot_786
Screenshot_787

### CODE REVIEW

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/MappingProfiles/CredentialsProfile.cs [hunk 1]
```diff
@@ -17,5 +19,13 @@ public CredentialsProfile()
                 => opts.MapFrom((_, dest) => dest.UserId))
             .ForMember(dest => dest.Type, opts
                 => opts.MapFrom((_, dest) => dest.Type));
+
+        CreateMap<MeetCredentialsDto, Credentials>()
+            .ForMember(dest => dest.UserId, opts 
+                => opts.MapFrom((_, dest) => dest.UserId))
```
Comment (Erbidium) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/MappingProfiles/CredentialsProfile.cs]: FIelds have same names, so they will probably be mapped automatically

**Review by Erbidium — CHANGES_REQUESTED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/GoogleMeetService.cs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        _context.Credentials.RemoveRange(credentials);
+        return _context.SaveChangesAsync();
+    }
+
+    public async Task CreateDefaultCredentials(string? userId = null)
+    {
+        var user = await _context.Users
+                       .Include(u => u.Credentials)
+                       .Include(u => u.Calendars)
+                       .FirstOrDefaultAsync(u => u.Uid == (userId ?? _userService.GetCurrentUserId()))
```
Comment (Erbidium) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/GoogleMeetService.cs]: I think you should compute user id outside of the query in order to not call GetCurrentUserId() each time

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/GoogleMeetService.cs [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        var user = await _context.Users
+                       .Include(u => u.Credentials)
+                       .Include(u => u.Calendars)
+                       .FirstOrDefaultAsync(u => u.Uid == (userId ?? _userService.GetCurrentUserId())) 
+                   ?? throw new KeyNotFoundException("User doesn't exist");
+        if (user.Credentials.All(c => c.Type != CredentialsType.Meet) && user.Calendars.Any())
+        {
+            var credentials = new Credentials
+            {
+                UserEmail = user.Calendars.First().ConnectedCalendar,
```
Comment (Erbidium) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/GoogleMeetService.cs:71]: Why first?

**Review by Erbidium — CHANGES_REQUESTED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/GoogleMeetService.cs (hunk 3 above)
Comment (YuraRiabov) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/GoogleMeetService.cs:71]: As a default value

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/GoogleMeetService.cs [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        _googleOAuthService = googleOAuthService;
+        _configuration = configuration;
+        _userService = userService;
+    }
+
+    public async Task<List<MeetCredentialsDto>> GetAvailableEmails()
+    {
+        var userId = _userService.GetCurrentUserId();
+        return await _context.Calendars
+            .Include(c => c.User)
```
[the remark this answers is not in the document]
Comment (YuraRiabov) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/GoogleMeetService.cs:31]: We need calendar's connected calendar value(this is email of connected calendar)

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/MappingProfiles/CredentialsProfile.cs (hunk 1 above)
Comment (YuraRiabov) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/MappingProfiles/CredentialsProfile.cs]: No, we map to destination from destination, not source

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/GoogleMeetService.cs (hunk 2 above)
Comment (YuraRiabov) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/GoogleMeetService.cs]: Fixed

File: backend/EasyMeets.Core/EasyMeets.Core.Common/DTO/GoogleMeet/EventTimeDto.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+﻿using Newtonsoft.Json;
+
+namespace EasyMeets.Core.Common.DTO.GoogleMeet;
+
+public class EventTimeDto
+{
+    [JsonProperty("dateTime")]
```
Comment (Erbidium) [inline backend/EasyMeets.Core/EasyMeets.Core.Common/DTO/GoogleMeet/EventTimeDto.cs:7]: Why we need this?

File: backend/EasyMeets.Core/EasyMeets.Core.Common/DTO/GoogleMeet/EntryPointDto.cs [hunk 6]
```diff
@@ -0,0 +1,9 @@
+﻿using Newtonsoft.Json;
+
+namespace EasyMeets.Core.Common.DTO.GoogleMeet;
+
+public class EntryPointDto
+{
+    [JsonProperty("uri")]
```
Comment (Erbidium) [inline backend/EasyMeets.Core/EasyMeets.Core.Common/DTO/GoogleMeet/EntryPointDto.cs:7]: Why we need this?

**Review by Erbidium — CHANGES_REQUESTED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/MappingProfiles/CredentialsProfile.cs (hunk 1 above)
Comment (YuraRiabov) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/MappingProfiles/CredentialsProfile.cs]: Fixed

**Review by Erbidium — APPROVED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/GoogleMeetService.cs (hunk 2 above)
Comment (apofis96) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/GoogleMeetService.cs]: spoiler: it will not call each time GetCurrentUserId (:

**Review by apofis96 — APPROVED** [review]

**Pull request merged**